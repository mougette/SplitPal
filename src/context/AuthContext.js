import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
