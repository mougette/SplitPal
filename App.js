import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import Ionicons from "react-native-vector-icons/Ionicons";
import Signin from './src/screens/Signin';
import SignupInfo from './src/screens/SignupInfo';
import Signup from "./src/screens/Signup";
import HomeScreen from './src/screens/HomeScreen';
import Friends from './src/screens/Friends';
import Split from './src/screens/Split';
import SplitGroup from './src/screens/SplitGroup';
import Groups from './src/screens/Groups';
import AddGroups from './src/screens/AddGroups';
import Profile from './src/screens/Profile';
import AddFriend from './src/screens/AddFriend';
import CameraScreen from './src/screens/CameraScreen';
import {Provider as AuthProvider} from './src/context/AuthContext.js';
import {Context as AuthContext} from './src/context/AuthContext';

const AuthStack = createStackNavigator();
function authFlow() {
  return (
    <AuthStack.Navigator>
      <AuthStack.Screen
        options={{headerShown: false}}
        name="Signin"
        component={Signin}
      />
      <AuthStack.Screen
        options={{headerShown: false}}
        name="Signup"
        component={Signup}
      />
        <AuthStack.Screen
            options={{headerShown: false}}
            name="SignupInfo"
            component={SignupInfo}
        />
    </AuthStack.Navigator>
  );
}

const GroupStack = createStackNavigator();
function groupFlow() {
    return (
        <GroupStack.Navigator>
            <GroupStack.Screen
                options={{headerShown: false}}
                name="Group"
                component={Groups}
            />
            <GroupStack.Screen
                options={{headerShown: false}}
                name="AddGroups"
                component={AddGroups}
            />
            <GroupStack.Screen name="SplitGroup" component={SplitGroup} />
        </GroupStack.Navigator>
    );
}
const FriendStack = createStackNavigator();
function friendFlow() {
  return (
    <FriendStack.Navigator>
      <FriendStack.Screen
        options={{headerShown: false}}
        name="Friend"
        component={Friends}
      />
      <FriendStack.Screen
        options={{headerShown: false}}
        name="Add Friends"
        component={AddFriend}
      />
      <FriendStack.Screen name="Split" component={Split} />
      <FriendStack.Screen name="CameraScreen" component={CameraScreen} />
    </FriendStack.Navigator>
  );
}

const Tab = createBottomTabNavigator();
function homeFlow() {
  return (
    <Tab.Navigator
      screenOptions={({route}) => ({
        tabBarIcon: ({focused, color, size}) => {
            let iconName;
            //determine which icon to display based on which tab use is in
            switch (route.name) {
                case "HomeScreen":
                    iconName = focused ? "home" : "home-outline";
                    break;
                case "Friends":
                    iconName = focused ? "person" : "person-outline";
                    break;
                case "Split":
                    iconName = focused ? "wallet" : "wallet-outline";
                    break;
                case "Groups":
                    iconName = focused ? "people" : "people-outline";
                    break;
                case "Profile":
                    iconName = focused ? "person-circle" : "person-circle-outline";
                    break;
            }
            return <Ionicons name={iconName} size={size} color={color} />;
        },
          tabBarActiveTintColor: "tomato",
          tabBarInactiveTintColor: "gray",
      })}
    >
      <Tab.Screen name="HomeScreen" component={HomeScreen} />
      <Tab.Screen name="Friends" component={friendFlow} />
      <Tab.Screen name="Groups" component={groupFlow} />
      <Tab.Screen name="Profile" component={Profile} />
    </Tab.Navigator>
  );
}

const Stack = createStackNavigator();
function App() {
  const {state} = React.useContext(AuthContext);
  console.log(state);
  return (
    <NavigationContainer>
      <Stack.Navigator>
        {state.token === null ? (
          <>
            <Stack.Screen
              options={{headerShown: false}}
              name="Auth"
              component={authFlow}
            />
          </>
        ) : (
          <Stack.Screen
            options={{headerShown: false}}
            name="Home"
            component={homeFlow}
          />
        )}
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default () => {
  return (
    <AuthProvider>
      <App />
    </AuthProvider>
  );
};
