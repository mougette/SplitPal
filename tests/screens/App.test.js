import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor} from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import App from '../../App'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("User info correct"),
  })
  );

jest.mock('expo-camera', () => { return "test"});
describe("App", () => {
  it('Login to App', async() => {

     const screen = render(<App/>);

     await waitFor(() => {
             fireEvent.press(screen.getByText("Login"))
             expect(screen.asFragment).toMatchSnapshot();
             })

  });
});