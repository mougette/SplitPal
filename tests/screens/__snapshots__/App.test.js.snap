// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Login to App 1`] = `undefined`;
